### ISSUE TITLE
Installing in v.0.7 : LoadError: symlink: file already exists (EEXIST)

### ISSUE THREAD

**Issue by colbec** [JuliaMath/SpecialFunctions.jl#50]
While attempting to update an existing `SpecialFunctions` in v 0.7 I hit a symlink error; v 0.6 also updates `SpecialFunctions` this morning but it builds without error.
So I removed the installation, checked that no cached version existed for `SpecialFunctions` in `.julia/v0.7/` and tried to reinstall. This leads to:
```
julia> Pkg.add("SpecialFunctions")
INFO: Installing SpecialFunctions v0.3.2
INFO: Building SpecialFunctions
INFO: No precompiled binaries found for your system. Building from scratch...
==========================[ ERROR: SpecialFunctions ]===========================

LoadError: LoadError: symlink: file already exists (EEXIST)
while loading /home/colin/.julia/v0.7/SpecialFunctions/deps/scratch.jl, in expression starting on line 8
while loading /home/colin/.julia/v0.7/SpecialFunctions/deps/build.jl, in expression starting on line 40

================================================================================

================================[ BUILD ERRORS ]================================

WARNING: SpecialFunctions had build errors.

 - packages with build errors remain installed in /home/colin/.julia/v0.7
 - build the package(s) and all dependencies with `Pkg.build("SpecialFunctions")`

```
Running Pkg.build just repeats the error.
It looks as though the install is looking for `libopenlibm.so`; at first this did not exist on my system, so I installed a version from my openSUSE repo and it now exists as `/usr/lib64/libopenlibm.so` but still the error persists.

```
julia> versioninfo()
Julia Version 0.7.0-DEV.1743
Commit 7e143bb333 (2017-09-09 20:19 UTC)
Platform Info:
  OS: Linux (x86_64-suse-linux)
  CPU: Intel(R) Core(TM) i5-4460  CPU @ 3.20GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, haswell)
Environment:

```

**Comment by H-M-H** [comment]
Actually I am also running into this issue with v0.6.0. It seems this is the offending line: https://github.com/JuliaMath/SpecialFunctions.jl/blob/dcbf013e44301a55df66a8a27abd630a12201020/deps/scratch.jl#L19-L20

**Comment by ararslan** [comment]
> at first this did not exist on my system

Had you built Julia using the system's libm then? Your `versioninfo()` shows Julia's libm as openlibm, which means that the build script should be using that.

Kind of mysterious that the `cp` call would be causing this, since the `remove_destination` option should remove the destination file if it exists before copying.

Thanks for the report, I'll look into it. In the meantime you can `Pkg.pin("SpecialFunctions", v"0.3.1")` to go back to the previous version.

**Comment by ararslan** [comment]
This should be fixed on master and in v0.3.3. Please let me know if you run into it again.
